libs/common/CMakeLists.txt

```diff
@@ -7,8 +7,17 @@ if(WIN32)
     # Locale only on windows
     set(_BoostComponents ${_BoostComponents} locale)
 endif()
-find_package(Boost 1.69 REQUIRED COMPONENTS ${_BoostComponents})
 
+set(BOOST_SOURCE TRUE)
+foreach(BOOST_PACKAGE IN LISTS _BoostComponents)
+    if(NOT TARGET Boost::${_BoostComponents})
+        set(BOOST_SOURCE FALSE)
+    endif()
+endforeach()
+
+if(NOT BOOST_SOURCE)
+    find_package(Boost 1.69 REQUIRED COMPONENTS ${_BoostComponents})
+endif()
```

Comment:
I think this code is generic enough to be put into a function. I suggest to create a macro `find_boost` such that it accepts `find_boost(1.69 REQUIRED COMPONENTS ${_BoostComponents})` just like `find_package` but first checks all components and then delegates to `find_package` if required.